tests/http/test_session.py

```diff
@@ -0,0 +1,62 @@
+import random
+from unittest import mock
+
+import pytest
+import requests
+from requests import adapters
+from requests_mock import Mocker
+
+from up42.http import session as up42_session
+
+some_url = "https://something.com"
+
+methods_and_calls = [
+    ("GET", requests.Session.get),
+    ("POST", requests.Session.post),
+    ("PUT", requests.Session.put),
+    ("PATCH", requests.Session.patch),
+    ("DELETE", requests.Session.delete),
+    ("OPTIONS", requests.Session.options),
+    ("HEAD", requests.Session.head),
+]
+
+authorization_value = "Bearer some-token"
+
+
+def set_token(request: requests.Request):
+    request.headers["Authorization"] = authorization_value
+    return request
+
+
+create_auth = mock.MagicMock(return_value=mock.MagicMock(side_effect=set_token))
```

Comment:
out of curiosity - why not define these inside the `create_session` fixture?
(since they are only used inside the fixture)

Reply:
No strong opinion here, moved it inside